api/k8s/v1/k8s.proto

```diff
@@ -419,6 +422,31 @@ message Deployment {
 
   map<string, string> labels = 4;
   map<string, string> annotations = 5;
+
+  message DeploymentStatus {
+    uint32 replicas = 1;
+    uint32 updated_replicas = 2;
+    uint32 ready_replicas = 3;
+    uint32 available_replicas = 4;
+    uint32 unavailable_replicas = 5;
+    message DeploymentCondition {
+      enum DeploymentConditionType {
+        UNSPECIFIED = 0;
+        UNKNOWN = 1;
+        AVAILABLE = 2;
+        PROGRESSING = 3;
+        REPLICA_FAILURE = 4;
+      }
+
+      DeploymentConditionType deployment_condition_type = 1;
+      google.protobuf.BoolValue condition_status = 2;
```

Comment:
This maps to https://pkg.go.dev/k8s.io/api@v0.20.5/core/v1#ConditionStatus right? so looks like it can have three values true, false and unknown, we might want to make this an enum too

Reply:
It looks like there might be plans to expand that to add other fields 
```
In the future, we could add other intermediate conditions, e.g. ConditionDegraded.
```
So enum might be our best bet